.pre-commit-config.yaml

```diff
@@ -9,7 +9,7 @@ repos:
         exclude: .github/repo_meta.yaml
     -   id: debug-statements
 -   repo: https://github.com/PyCQA/isort
-    rev: 5.10.1
+    rev: 5.12.0
```

Comment:
5.12.0 fails the CI because it has removed python 3.7 support.
some ideas that we could try:

1. downgrade to 5.11.*
2. check if pre commit allows us to conditionally run the isort depending on the python env

Reply:
Downgrading to 5.11 will fail on install (because precommit doesn't let you build wheels, so will build from sdist and fail again like earlier). 

Consider not running pre-commit on 3.7 at all?

Reply:
thanks @hauntsaninja for the suggestion, @sfc-gh-jdu could you do the update?

Reply:
done